fs: zms: fix no-cache + no-double-write

JordanYates opened the pull request:
Fix a compiler error when `CONFIG_ZMS_NO_DOUBLE_WRITE=y` and `CONFIG_ZMS_LOOKUP_CACHE=n`.

Fixes #95614

JordanYates committed:
fs: zms: fix no-cache + no-double-write

Fix a compiler error when `CONFIG_ZMS_NO_DOUBLE_WRITE=y` and
`CONFIG_ZMS_LOOKUP_CACHE=n`.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
tests: fs: zms: test `CONFIG_ZMS_NO_DOUBLE_WRITE`

Enable tests for `CONFIG_ZMS_NO_DOUBLE_WRITE=y`.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates commented on subsys/fs/zms/zms.c:
```diff
@@ -1498,12 +1498,19 @@ ssize_t zms_write(struct zms_fs *fs, uint32_t id, const void *data, size_t len)
+			goto no_cached_entry;
+		} else {
+			/* skip delete entry for non-existing entry */
+			return 0;
```
In my head this is not correct, since I assume it is possible for an entry to not be in the cache when the delete is requested.
However I have been unable to create a test-case that generates that behaviour, and in any case I assume that would mean the larger `wlk_addr == ZMS_LOOKUP_CACHE_NO_ADDR` conditional is also incorrect, since we couldn't assume that the entry doesn't exist just because its not in the cache.

rghaddab replied:
@JordanYates which test is failing ?
Is it the test_zms_cache_collision ?
I will test locally to verify this

JordanYates committed:
fs: zms: fix delete non-existent with cache

Fix deleting a key that doesn't exist from writing a new entry to the
filesystem when `CONFIG_ZMS_NO_DOUBLE_WRITE` and
`CONFIG_ZMS_LOOKUP_CACHE` are enabled.

Signed-off-by: Jordan Yates <jordan@embeint.com>
